pkg/controllers/deprovisioning/controller.go

```diff
@@ -249,18 +256,18 @@ func (c *Controller) launchReplacementNodes(ctx context.Context, action Command,
 	})
 	multiErr := multierr.Combine(errs...)
 	if multiErr != nil {
-		c.cluster.UnmarkForDeletion(nodeNamesToRemove...)
-		return multierr.Combine(c.setNodesUnschedulable(ctx, false, nodeNamesToRemove...),
+		c.cluster.UnmarkForDeletion(lo.Map(action.candidatesToRemove, func(c *Candidate, _ int) string { return c.Name() })...)
```

Comment:
Any reason you want to assign this to a variable rather than just take care of it in-place? Not sure that we really need to assign this to a variable here